docs/source/guides/faq.md

```diff
@@ -0,0 +1,121 @@
+
+For errors not prevented by the type system, it's helpful to know what exact queries were made, and with what variables. [Apollo Engine](https://www.apollographql.com/engine) is a tool that does exactly this. It can help discover and reproduce errors by showing the exact conditions in which the error occurred.
+
+## Moving a feature to GraphQL
+
+> People in this phase have decided to use GraphQL in production. They don't want to immediately refactor the APIs or apps. They want to move a single feature over to GraphQL to learn how to use it and monitor it in production.
+
+#### How do I set up authentication/authorization for my GraphQL schema?
+
+Authentication and authorization are important topics to discuss with any API. GraphQL provides a very granular approch to handling these topics. But don't worry! If an API being consumed by GraphQL already has authorization built-in, it may be possible to ignore it completely! Read more in the [access control guide](./access-control.html).
+
+#### How can I secure my schema from malicious or expensive queries?
+
+Public APIs of any kind need some kind of safeguards against malicious queries. Since GraphQL allows for recursive queries, it wouldn't be hard to create a query that is overly complicated and acts as a DoS attack, even by accident. There are multiple ways to prevent something like this from happening, from complexity limiting to query depth limiting. Read the [guide on security](./security.html) to learn more.
+
+#### What kinds of cache should I setup?
+
+GraphQL can be cached in multiple places. It can be cached on the client, preventing multiple queries from being made when not necessary.
+
+Cache can also be setup at the schema level. Whole-query caching, partial-query caching, and cache backed by a CDN can all be used to lower response times, and make a GraphQL schema as performant as possible.
+
+Whole-query and CDN caches are most useful when an API receives many of the same queries. This commonly happens with public data, like content on pages of a site. Regardless of whether the API is used for public data or not, these caches almost always provide large performance benefits, and are highly recommended. You can read more about how to set up whole-query and CDN caching with `apollo-server` 2.0 [here](https://www.apollographql.com/docs/guides/performance.html).
+
+Partial query caching can be achieved by caching the responses from underlying services with something like Redis or Memcache. With this strategy, even if two queries look completely different from one another, if there is any duplication of data fetched, those results can be shared, preventing unnecessary traffic. The [`RESTDataSource`](https://www.apollographql.com/docs/apollo-server/features/data-sources.html) does this automatically if the appropriate `cache-control` headers are present in REST responses.
+
+#### How can I monitor the health of my GraphQL schema?
```

Comment:
it would be worth mentioning Engine Slack integration - alerts and daily reports